crates/fluvio-types/src/event.rs

```diff
@@ -9,47 +9,69 @@ const DEFAULT_EVENT_ORDERING: Ordering = Ordering::SeqCst;
 #[deprecated(since = "0.2.5", note = "use StickyEvent instead")]
 pub use StickyEvent as SimpleEvent;
 
+#[derive(Clone)]
 pub struct StickyEvent {
+    inner: Arc<StickyInner>,
+}
+
+struct StickyInner {
     flag: AtomicBool,
     event: Event,
 }
 
+impl StickyInner {
+    fn is_set(&self) -> bool {
+        self.flag.load(DEFAULT_EVENT_ORDERING)
+    }
+}
+
 impl StickyEvent {
+    #[deprecated(since = "0.2.6", note = "use 'StickyEvent::new()' instead")]
     pub fn shared() -> Arc<Self> {
-        Arc::new(Self {
-            flag: AtomicBool::new(false),
-            event: Event::new(),
-        })
+        Arc::new(Self::new())
+    }
+
+    pub fn new() -> Self {
+        Self {
+            inner: Arc::new(StickyInner {
+                flag: AtomicBool::new(false),
+                event: Event::new(),
+            }),
+        }
     }
 
     // is flag set
     pub fn is_set(&self) -> bool {
```

Comment:
use `Deref` trait rather than duplicating method

Reply:
Alright, I just tried this out and I'm getting the following error:

```
error[E0446]: private type `StickyInner` in public interface
  --> crates/fluvio-types/src/event.rs:75:5
   |
18 | struct StickyInner {
   | ------------------ `StickyInner` declared as private
...
75 |     type Target = StickyInner;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak private type
```

I don't really want to expose the `StickyInner` type, can we leave it as-is?